src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -201,10 +212,19 @@ private NetworkPeerConnectionParameters CreateNetworkPeerConnectionParameters()
         /// <returns>When criteria is met returns <c>true</c>, to allow connection.</returns>
         private (bool successful, string reason) AllowClientConnection(TcpClient tcpClient)
         {
+            var clientLocalEndPoint = tcpClient.Client.LocalEndPoint as IPEndPoint;
+
+            var peer = this.peerAddressManager.FindPeer(clientLocalEndPoint);
```

Comment:
Should this be `FindPeersByIp` instead with a loop through the returned results? If a peer connects from a different ephemeral port (highly likely) it will not be found by `FindPeer`.

Reply:
True, cause peer banning sets the banned time on all the peers with the same IP, regardless of port.